main.go

```diff
@@ -98,7 +103,7 @@ func startRPCServer(db *v1.DB, kubeConfig *v1.Config) {
 	api.RegisterWorkflowServiceServer(s, server.NewWorkflowServer())
 	api.RegisterSecretServiceServer(s, server.NewSecretServer())
 	api.RegisterNamespaceServiceServer(s, server.NewNamespaceServer())
-	api.RegisterAuthServiceServer(s, server.NewAuthServer())
+	api.RegisterAuthServiceServer(s, server.NewAuthServer(namespaces, config))
```

Comment:
@Vafilor I wonder if a better pattern would be to cache `SystemConfig` when `GetSystemConfig()` is called the first time and subsequent calls to `GetSystemConfig()` would use the cached value. `GetSytemConfig()` is called in multiple places in code so this will make things more efficient across the board.

Reply:
@rushtehrani Pushed up a change for this. Let me know what you think.

Reply:
@Vafilor looks good, you don't need to pass it in anymore, you can just call `GetSystemConfig()` in `NewAuthServer`.

Reply:
@rushtehrani 

Roger that - this answers a question in a comment below.